src/main/java/spireMapOverhaul/zones/goldencurse/RewardPatches.java

```diff
@@ -0,0 +1,103 @@
+package spireMapOverhaul.zones.goldencurse;
+
+import basemod.abstracts.CustomReward;
+import com.evacipated.cardcrawl.modthespire.lib.SpireField;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch2;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.potions.AbstractPotion;
+import com.megacrit.cardcrawl.relics.AbstractRelic;
+import com.megacrit.cardcrawl.rewards.RewardItem;
+import spireMapOverhaul.patches.ZonePatches;
+
+import static com.megacrit.cardcrawl.dungeons.AbstractDungeon.player;
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+
+
+public class RewardPatches {
+    @SpirePatch2(
+            clz = RewardItem.class,
+            method = SpirePatch.CLASS
+    )
+    public static class RewardItemFields {
+        public static SpireField<Integer> cost = new SpireField<>(() -> 0);
+    }
+
+    @SpirePatch2(
+            clz = RewardItem.class,
+            method = SpirePatch.CONSTRUCTOR,
+            paramtypez = {}
+
+    )
+    @SpirePatch2(
+            clz = RewardItem.class,
+            method = SpirePatch.CONSTRUCTOR,
+            paramtypez = {AbstractRelic.class}
+
+    )
+    @SpirePatch2(
+            clz = RewardItem.class,
+            method = SpirePatch.CONSTRUCTOR,
+            paramtypez = {RewardItem.class, RewardItem.RewardType.class}
+
+    )
+    @SpirePatch2(
+            clz = RewardItem.class,
+            method = SpirePatch.CONSTRUCTOR,
+            paramtypez = {AbstractPotion.class}
+
+    )
+    public static class RewardItemConstructors {
+        public static void Postfix(RewardItem __instance) {
+            if (!(ZonePatches.currentZone() instanceof GoldenCurseZone)) {
+                return;
+            }
+            int goldAmt = GoldenCurseZone.getRewardCost(__instance);
+
+            RewardItemFields.cost.set(__instance, goldAmt);
+            if (goldAmt > 0) {
+                __instance.text = __instance.text + TEXT[0] + goldAmt;
```

Comment:
This is liable to go over the text length limit or be replaced in certain situations.

How about we just render a gold icon and the number next to it maybe in red) in the bottom right corner of the reward?
Here's an example where we do something like this
https://github.com/erasels/RoR-StS-Mod/blob/7184c76181b3be0c573fc800f9037b70fd9f18e2/RoRStS/src/main/java/riskOfSpire/rewards/ExpensiveLinkedReward.java#L59

Reply:
makes sense. ill change it soon.